storage/src/mmr/mutable.rs

```diff
@@ -0,0 +1,512 @@
+//! An in-memory authenticatable & updatable key-value store.
+//!
+//! A _mutable MMR_ is an authenticatable and updatable key-value store based on an [Mmr] over all
+//! updates called the _update tree_, and an authenticatable [Bitmap] that indicates which leaves in
+//! the update tree correspond to the latest value of each key. For example, if the same key K was
+//! updated at positions X, Y, and Z in that order, then the bitmap will store 0s for the bits
+//! corresponding to X and Y, and a 1 for the bit corresponding to Z. The root hash of the mutable
+//! MMR is the result of hashing together the roots of these two structures.
+
+use crate::mmr::{
+    bitmap::Bitmap, hasher::Hasher, iterator::leaf_pos_to_num, mem::Mmr,
+    verification::KeyValueProof, Error,
+};
+use commonware_cryptography::Hasher as CHasher;
+use commonware_utils::Array;
+use std::{collections::HashMap, marker::PhantomData};
+use tracing::error;
+
+/// A record representing one key/value update or deletion.
+pub struct UpdateRecord<K: Array, V: Array> {
+    /// The position of the leaf representing this record in the updates tree, or Bitmap::DELETED if
```

Comment:
In this comment, you refer to `Bitmap::DELETED` but I think it is defined in this file?

Reply:
oops correct, Changed to MutableMmr::DELETED and linkified it.